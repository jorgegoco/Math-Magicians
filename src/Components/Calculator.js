import React, { Component } from 'react';
import calculate from '../logic/calculate';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
  }

  handleClick(e) {
    const buttonValue = e.target.className;
    return this.setState((obj) => calculate(obj, buttonValue));
  }

  // onChangeHandler(e) {  
  // }

  render() {
    const { total, next, operation } = this.state;
    const result = `${total}${operation}${next}`.replace(/null/g, '');
    const trimedResult = result.replace(/undefined/g, '');
    return (
      <div className="calculator-container">
        <div className="output">
          {trimedResult}
          {/* <input onChange={onChangeHandler} /> */}
        </div>
        <button className="ac" type="button" onClick={this.handleClick}>AC</button>
        <button className="plus-minus" type="button" onClick={this.handleClick}>+/-</button>
        <button className="percentage" type="button" onClick={this.handleClick}>%</button>
        <button className="division" type="button" onClick={this.handleClick}>/</button>
        <button className="seven" type="button" onClick={this.handleClick}>7</button>
        <button className="eight" type="button" onClick={this.handleClick}>8</button>
        <button className="nine" type="button" onClick={this.handleClick}>9</button>
        <button className="product" type="button" onClick={this.handleClick}>*</button>
        <button className="four" type="button" onClick={this.handleClick}>4</button>
        <button className="five" type="button" onClick={this.handleClick}>5</button>
        <button className="six" type="button" onClick={this.handleClick}>6</button>
        <button className="minus" type="button" onClick={this.handleClick}>-</button>
        <button className="one" type="button" onClick={this.handleClick}>1</button>
        <button className="two" type="button" onClick={this.handleClick}>2</button>
        <button className="three" type="button" onClick={this.handleClick}>3</button>
        <button className="plus" type="button" onClick={this.handleClick}>+</button>
        <button className="zero" type="button" onClick={this.handleClick}>0</button>
        <button className="dot" type="button" onClick={this.handleClick}>.</button>
        <button className="equal" type="button" onClick={this.handleClick}>=</button>
      </div>
    );
  }
}

export default Calculator;
