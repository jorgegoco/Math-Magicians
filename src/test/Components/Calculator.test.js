import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../../Components/Calculator';

describe('Calculator component test', () => {
  test('snapshot renders', () => {
    const calc = renderer
      .create(<Calculator />)
      .toJSON();
    expect(calc).toMatchSnapshot();
  });

  test('user interaction', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByTestId('testResult')).toHaveTextContent(/3/i);
  });

  test('matches element', () => {
    render(<Calculator />);
    expect(screen.getByText(/math/)).toBeInTheDocument();
  });
});
