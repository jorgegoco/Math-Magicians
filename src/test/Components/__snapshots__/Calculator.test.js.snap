// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component test snapshot renders 1`] = `
<section
  className="calculatorSection"
>
  <h1>
    Let's do some math!
  </h1>
  <div
    className="calculator-container"
  >
    <div
      className="output"
      data-testid="testResult"
    >
      0
    </div>
    <button
      className="ac"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="plus-minus"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="percentage"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="division"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="seven"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="eight"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="nine"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="product"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="four"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="five"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="six"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="minus"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="one"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="two"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="three"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="plus"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;
